### PR TITLE
pkg/keepalive: revoke lease if we quit normally

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve? 
https://github.com/pingcap/dm/issues/525

### What is changed and how it works?
Revoke lease before we quit normally once we put a key with lease successfully.

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: pkg/ha/keepalive.go [hunk 1]
```diff
@@ -104,14 +106,17 @@ func KeepAlive(ctx context.Context, cli *clientv3.Client, workerName string, kee
...
 			log.L().Info("ctx is canceled, keepalive will exit now")
-			ctx, cancel := context.WithTimeout(cli.Ctx(), etcdutil.DefaultRequestTimeout)
-			cli.Revoke(ctx, lease.ID)
-			cancel()
 			return nil
 		}
 	}
 }
 
+func revokeLease(ctx context.Context, cli *clientv3.Client, id clientv3.LeaseID) (*clientv3.LeaseRevokeResponse, error) {
```
Comment (csuzhangxc) [inline pkg/ha/keepalive.go]: this `ctx` is not used.

File: pkg/ha/keepalive.go [hunk 2]
```diff
@@ -104,14 +106,17 @@ func KeepAlive(ctx context.Context, cli *clientv3.Client, workerName string, kee
...
+	ctx, cancel := context.WithTimeout(cli.Ctx(), etcdutil.DefaultRevokeLeaseTimeout)
```
Comment (csuzhangxc) [inline pkg/ha/keepalive.go]: how about adding a comment to highlight _we must ensure `cli.Ctx()` not done when exiting_ (do not set `Context` when creating `cli` or do not cancel this `Context` or it's parent context).

Comment (lichunzhu) [reply pkg/ha/keepalive.go]: Addressed in https://github.com/pingcap/dm/pull/548/commits/6e69519b2d12e077049442d350f92c75ff44fc43

**Pull request merged**